### PR TITLE
Supports binary data

### PR DESCRIPTION
**Pull request by haginaga**
This will support binary data.

```
@app.route('/upload', methods=['POST'],
        content_types=['application/octet-stream'],
        binary_support=True)
def upload:
    # Body is binary data.
    body = app.current_request.raw_body
    # Use boto3 s3 client
    s3.put_object(
        Bucket=S3_BUCKET,
        Key=KEY,
        Body=body,
        ContentType='application/octet-stream'
    )
    return {}
```
https://aws.amazon.com/jp/about-aws/whats-new/2016/11/binary-data-now-supported-by-api-gateway/

### CODE REVIEW

**Comment by mihow** [comment]
Will this cover binary responses as well? For example, a dynamically generated image.

**Comment by haginaga** [comment]
@mihow 
Yes. 
However a client needs to specify the `content_types` specified in the accept header.
For example, 
when specified `content_types=['image/png'] binary_support=True` in the `@app.route()` entry,
a client needs to attach an 'Accept: image/png' header in request.

**Comment by mihow** [comment]
Thank you @haginaga! I will try this.

**Comment by artkay** [comment]
Any news on merging this?

**Comment by jamesls** [comment]
Awesome, thanks for the pull request.  Taking a look at this now.

File: chalice/deployer.py [hunk 1]
```diff
@@ -31,7 +31,8 @@
         "Sid": "",
         "Effect": "Allow",
         "Principal": {
-            "Service": "lambda.amazonaws.com"
+            "Service": "lambda.amazonaws.com",
+            "Service": "apigateway.amazonaws.com",
```
Comment (jamesls) [inline chalice/deployer.py]: Why was apigateway added here?

Comment (haginaga) [inline chalice/deployer.py]: @jamesls 
Thank you for the code review.

As you pointed out. Sorry. I'll fix it soon.

**Comment by jamesls** [comment]
@haginaga I'd like to this merged so I'm happy to take this PR and incorporate my remaining feedback.

Let me know if you're working on this so we don't duplicate effort, but otherwise I'll start making changes.

**Comment by haginaga** [comment]
@jamesls I'm sorry I did not reply.
I have been busy here for several weeks and I can not code it. You may start changing.

**Comment by ogrodnek** [comment]
Would love to see this merged, if possible...

**Comment by knoxilla** [comment]
Looking forward to getting this merged.    Thanks!

**Comment by haginaga** [comment]
I post new PR #286 on Chalice version 0.7.0.
I close this PR.

**Pull request closed without merging**